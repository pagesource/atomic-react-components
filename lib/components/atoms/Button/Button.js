// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import classnames from 'classnames';

import { getLabelFromChildren } from '../../../core/utils/elementLocator';
import styles from './Button.style';

type Props = {
  className?: string,
  children: Node,
  inheritedStyles?: string,
  type?: string,
  primary?: boolean,
  secondary?: boolean,
  tertiary?: boolean,
  disabled?: boolean,
  ariaLabel?: string,
  elementLocator?: string,
  icon?: Boolean | String,
  iconImage?: String,
};

const Button = ({
  className,
  children,
  inheritedStyles,
  type,
  primary,
  secondary,
  tertiary,
  disabled,
  ariaLabel,
  elementLocator,
  icon,
  iconImage,
  ...others
}: Props): Node => (
  <button
    aria-label={ariaLabel || null}
    className={classnames(className, { 'icon icon-image': iconImage })}
    disabled={disabled}
    type={type}
    data-locator={elementLocator || `button-${type}-${getLabelFromChildren(children)}`}
    {...others}
  >
    {children}
  </button>
);

Button.defaultProps = {
  inheritedStyles: '',
  type: 'button',
  primary: false,
  secondary: false,
  tertiary: false,
  disabled: false,
  ariaLabel: '',
  className: '',
  elementLocator: '',
  icon: false,
  iconImage: '',
};

export default styled(Button)`
  ${styles};
`;
export { Button as ButtonVanilla };
