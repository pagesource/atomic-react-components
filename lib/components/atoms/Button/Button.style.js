import { css } from 'styled-components';

const ButtonStyles = css`
  border: none;
  border-radius: 0;
  position: relative;
  min-height: 48px;
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  ${props => `
      ${props.theme.colors.CTA_PRIMARY};
      ${props.theme.typography.FONT_CTA};
      padding: 0 ${props.theme.spacing.UNIT_2};
  `};
  ${props => props.secondary && `${props.theme.colors.CTA_SECONDARY}`};
  ${props =>
    props.tertiary
      ? `
    ${props.theme.colors.CTA_TERTIARY};
    ${props.theme.border.CTA_TERTIARY};
  `
      : ``};
  ${props => props.fullWidth && ` width: 100% ;`};

  &:focus:not(.focus-visible) {
    outline: none;
  }

  &.icon {
    border: none;
    background-color: transparent;
    text-indent: -1000px;
    overflow: hidden;
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    min-width: 20px;
    min-height: 20px;
    padding: 0;

    &.icon-image {
      background-image: url('${props => props.iconImage}');
    }
  }

  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default ButtonStyles;
