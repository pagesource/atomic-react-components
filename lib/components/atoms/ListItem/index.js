// @flow
export { default } from './ListItem';
export { ListItemVanilla } from './ListItem';
