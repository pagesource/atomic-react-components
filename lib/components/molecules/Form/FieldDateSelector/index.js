// @flow
export { default } from './FieldDateSelector';
export { FieldDateSelectorVanilla } from './FieldDateSelector';
