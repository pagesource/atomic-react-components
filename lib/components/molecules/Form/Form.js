// @flow
import React from 'react';
import { useForm } from 'react-hook-form';
import type { Node } from 'react';
import { buildSchema } from '../../../core/utils/buildValidationSchema';
import Debug from './Debug';
import FieldInput from './FieldInput';
import FieldSelect from './FieldSelect';
import FieldTextarea from './FieldTextarea';
import FieldRadioGroup from './FieldRadioGroup';
import FieldCheckbox from './FieldCheckbox';
import FieldToggle from './FieldToggle';
import FieldFileInput from './FieldFileInput';
import DDFileInput from './DragDropInput';

type Props = {
  initialValues?: any,
  children: Node,
  onSubmit: Function,
  debug?: boolean,
  validationSchema?: Object | null,
  validate?: String,
  yupValidationSchema?: Function | null,
};

const Form = ({
  onSubmit,
  initialValues,
  children,
  debug,
  validate,
  validationSchema,
  yupValidationSchema,
}: Props) => {
  const getSchema = () => {
    if (yupValidationSchema) return yupValidationSchema;
    return validationSchema ? buildSchema(validationSchema) : null;
  };
  const { handleSubmit, register, errors, setValue } = useForm({
    mode: validate,
    validationSchema: getSchema(),
    defaultValues: initialValues,
  });

  // $flow-disable-line  eslint-disable-line
  const childrenWithProps = children[0].props.children.map((child) => {
    return React.cloneElement(child, { register, errors, setValue });
  });

  // $flow-disable-line  eslint-disable-line
  childrenWithProps.push(children[1]);
  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      {childrenWithProps}
      {debug && <Debug />}
    </form>
  );
};

Form.Checkbox = FieldCheckbox;

Form.Input = FieldInput;

Form.RadioGroup = FieldRadioGroup;

Form.Select = FieldSelect;

Form.Textarea = FieldTextarea;

Form.Toggle = FieldToggle;

Form.FieldFile = FieldFileInput;

Form.DDFile = DDFileInput;

Form.defaultProps = {
  debug: false,
  initialValues: null,
  validationSchema: null,
  yupValidationSchema: null,
  validate: 'onSubmit',
};

export default Form;
