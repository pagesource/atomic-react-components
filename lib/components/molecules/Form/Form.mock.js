import { action } from '@storybook/addon-actions';

const userFormProps = {
  initialValues: {
    email: '',
    username: '',
    typeOfUser: 'Admin',
    comment: '',
    ageGroup: '',
    termsAndConditions: false,
  },
  debug: false,
  action: '/',
  validationSchema: {
    username: {
      max: [15, 'Max limit reached'],
      required: 'User name Required',
    },
    email: {
      max: [30, 'Max reached'],
      email: 'Invalid email format',
      required: 'Required',
    },
    comment: {
      max: [100, 'Comment should not be more than 100 characters'],
      required: 'Please enter a comment',
    },
    ageGroup: {
      required: 'Please select an Age group',
    },
    termsAndConditions: {
      type: 'bool',
      test: [
        'termsAndConditions',
        'please accept terms and conditions!!!',
        (value) => value === true,
      ],
    },
  },
  onSubmit: action('On Submit Fired'),
};

const userFormSelectProps = {
  options: ['Admin', 'Read Only'],
  name: 'typeOfUser',
  label: 'Type of user',
};

const userFormNameInputProps = {
  name: 'username',
  type: 'text',
  label: 'User Name',
  validationInstructions: 'Enter valid username',
};

const fileMultipleInputProps = {
  name: 'fileuploadMulti',
  localeText: {
    btnTextPrimary: 'Choose Files',
    btnTextSecondary: 'Upload...',
    dragHelperText: 'Drop Files Here...',
  },
  multiple: true,
};
const fileSingleInputProps = {
  name: 'fileupload',
  localeText: {
    btnTextPrimary: 'Choose File',
    btnTextSecondary: 'Upload...',
    dragHelperText: 'Drop File Here...',
  },
};
const fileDDInputProps = {
  name: 'fileuploadDD',
  localeText: {
    btnTextPrimary: 'Choose File',
    btnTextSecondary: 'Upload...',
    dragHelperText: 'Drop File Here...',
  },
};

const userFormEmailProps = {
  type: 'email',
  name: 'email',
  label: 'Email',
  validationInstructions: 'Enter valid email',
};

const userFormCommentProps = {
  name: 'comment',
  label: 'Comment',
  placeholder: 'Enter your comment here...',
};

const userFormAgeGroupProps = {
  name: 'ageGroup',
  radioGroupArray: [
    {
      value: '30-',
      label: '< 30',
    },
    {
      value: '30-60',
      label: '30-60',
    },
    {
      value: '60+',
      label: '> 60',
    },
  ],
};

const userFormTnCProps = {
  name: 'termsAndConditions',
  label: 'I accept terms & conditions',
};

const userFormToggleProps = {
  name: 'test_1',
  id: 'checkbox_test_1',
  value: 'off',
  label: 'Toggle',
  indicatorOn: 'On',
  indicatorOff: 'Off',
  disabled: false,
  intialToggleState: false,
  displayIndicators: true,
};

export {
  userFormProps,
  fileMultipleInputProps,
  fileSingleInputProps,
  userFormSelectProps,
  userFormNameInputProps,
  userFormEmailProps,
  userFormCommentProps,
  userFormAgeGroupProps,
  userFormTnCProps,
  userFormToggleProps,
  fileDDInputProps,
};
