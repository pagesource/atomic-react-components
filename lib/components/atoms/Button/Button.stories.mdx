import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { primaryButton, primaryButtonDisabled, secondaryButton, secondaryButtonDisabled, customButtonLocator} from './Button.mock';
import Button, { ButtonVanilla } from '.';

<Meta title="Atoms/Buttons" component={Button} />

### Vanila  Button

<Preview>
  <Story name="Vanila Button">
   <ButtonVanilla {...primaryButton}>Submit</ButtonVanilla>
  </Story>
   <Story name="Vanila Disabled Button">
   <ButtonVanilla {...primaryButtonDisabled}>Submit</ButtonVanilla>
  </Story>
</Preview>


### Primary  Button

<Preview>
  <Story name="Primary Styled Button">
   <Button {...primaryButton}>Submit</Button>
  </Story>
   <Story name="Primary Styled Disabled Button">
   <Button {...primaryButtonDisabled}>Submit</Button>
  </Story>
</Preview>


### Secondary Button

<Preview>
  <Story name="Secondary Styled Button">
   <Button {...secondaryButton}>Submit</Button>
  </Story>
   <Story name="Secondary Styled Disabled Button">
   <Button {...secondaryButtonDisabled}>Submit</Button>
  </Story>
</Preview>

### custom Button Locator

<Preview>
  <Story name="Custom  Button">
   <Button {...customButtonLocator}>Submit</Button>
  </Story>
</Preview>

## Props ðŸ”§

<Props of={ButtonVanilla} />
