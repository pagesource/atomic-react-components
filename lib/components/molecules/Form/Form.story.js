// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  userFormProps,
  userFormSelectProps,
  fileMultipleInputProps,
  fileSingleInputProps,
  userFormNameInputProps,
  userFormEmailProps,
  userFormCommentProps,
  userFormAgeGroupProps,
  userFormDateSelectorProps,
  userFormTnCProps,
  userFormToggleProps,
} from './Form.mock';

// Import Vanilla Component to display markup and props
import Form from './Form';
import Button from '../../atoms/Button';
// Import Styled Component to showcase variations

storiesOf('Molecules/Form', module)
  .addParameters({ jest: ['Form', 'Button'] })
  .add('Default', () => (
    <Form {...userFormProps}>
      <div>
        <Form.Select {...userFormSelectProps} />
        <Form.Input {...userFormNameInputProps} />
        <Form.Input {...userFormEmailProps} />
        <Form.FieldFile {...fileMultipleInputProps} />
        <Form.FieldFile {...fileSingleInputProps} />
        <Form.DDFile {...fileSingleInputProps} />
        <Form.RadioGroup {...userFormAgeGroupProps} />
        <Form.DateSelector {...userFormDateSelectorProps} />
        <Form.Checkbox {...userFormTnCProps} />
        <Form.Textarea {...userFormCommentProps} />
        <Form.Toggle {...userFormToggleProps} />
      </div>
      <Button primary type="submit">
        Submit
      </Button>
    </Form>
  ));
