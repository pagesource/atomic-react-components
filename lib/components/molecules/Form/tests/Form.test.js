import React from 'react';
import * as Yup from 'yup';
import 'mutationobserver-shim';

import { shallowWithTheme } from '../../../../../__mock__/themeMock';
import Form from '..';
import { ButtonVanilla } from '../../../atoms/Button';

global.MutationObserver = window.MutationObserver;

describe('Form Component', () => {
  const userFormProps = {
    initialValues: {
      email: '',
      username: '',
      typeOfUser: '',
    },
    debug: false,
    action: '/',
    validationSchema: {
      username: {
        max: [15, 'Max limit reached'],
        required: 'User name Required',
      },
      email: {
        max: [30, 'Max reached'],
        email: 'Invalid email format',
        required: 'Required',
      },
    },
  };

  const SignupSchema = Yup.object().shape({
    username: Yup.string().min(2, 'Too Short!').max(50, 'Too Long!').required('Required'),
  });

  const userFormNameInputProps = {
    name: 'username',
    type: 'text',
    label: 'User Name',
  };

  const userFormSelectProps = {
    options: ['Admin', 'Read Only'],
    name: 'typeOfUser',
    label: 'Type of user',
    validateOnBlur: true,
  };

  test('should render form correctly', () => {
    const FormTag = (
      <Form {...userFormProps}>
        <div>
          <Form.Select {...userFormSelectProps} />
          <Form.Input {...userFormNameInputProps} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    const Wrapper = shallowWithTheme(FormTag);
    expect(Wrapper).toMatchSnapshot();
  });

  test('should render form field with yupValidationSchema', () => {
    const FormField = (
      <Form {...userFormProps} yupValidationSchema={SignupSchema}>
        <div>
          <Form.Input {...userFormNameInputProps} />
          <Form.Select {...userFormSelectProps} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    const Wrapper = shallowWithTheme(FormField);
    expect(Wrapper).toMatchSnapshot();
  });

  test('should render form field with no validation', () => {
    const formProps = { ...userFormProps };
    delete formProps.validationSchema;
    const FormField = (
      <Form {...formProps}>
        <div>
          <Form.Input {...userFormNameInputProps} />
          <Form.Select {...userFormSelectProps} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    const Wrapper = shallowWithTheme(FormField);
    expect(Wrapper).toMatchSnapshot();
  });

  test('getSchema should return null if no params are passed', () => {
    const formProps = { ...userFormProps };
    delete formProps.validationSchema;
    const FormField = (
      <Form {...formProps}>
        <div>
          <Form.Input {...userFormNameInputProps} />
          <Form.Select {...userFormSelectProps} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    const Wrapper = shallowWithTheme(FormField);
    expect(Wrapper.instance()).toBeNull();
  });
});
