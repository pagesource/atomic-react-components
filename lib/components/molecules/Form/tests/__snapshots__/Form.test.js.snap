// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should render input field with no validation 1`] = `
<Formik
  initialValues={
    Object {
      "email": "",
      "typeOfUser": "",
      "username": "",
    }
  }
  validate={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={null}
>
  <Component />
</Formik>
`;

exports[`Form Component should render input field with yupValidationSchema 1`] = `
<Formik
  initialValues={
    Object {
      "email": "",
      "typeOfUser": "",
      "username": "",
    }
  }
  validate={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "username",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "username": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "max": true,
            "min": true,
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
>
  <Component />
</Formik>
`;

exports[`Form Component should render input text field 1`] = `
<Formik
  initialValues={
    Object {
      "email": "",
      "typeOfUser": "",
      "username": "",
    }
  }
  validate={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "email",
        "username",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "email": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "max": true,
            "required": true,
            "undefined": false,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "username": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "max": true,
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
>
  <Component />
</Formik>
`;

exports[`Form Component should render select field 1`] = `
<Formik
  initialValues={
    Object {
      "email": "",
      "typeOfUser": "",
      "username": "",
    }
  }
  validate={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "email",
        "username",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "email": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "max": true,
            "required": true,
            "undefined": false,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "username": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "max": true,
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
>
  <Component />
</Formik>
`;

exports[`Form Component should render select tag correctly 1`] = `
<Formik
  initialValues={
    Object {
      "email": "",
      "typeOfUser": "",
      "username": "",
    }
  }
  validate={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "email",
        "username",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "email": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "max": true,
            "required": true,
            "undefined": false,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "username": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "max": true,
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
>
  <Component />
</Formik>
`;
